sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/AvroCarbonWriter.java

```diff
@@ -823,6 +887,36 @@ public void write(Object object) throws IOException {
     }
   }
 
+  /**
+   * Load data of all avro files at given location iteratively.
+   *
+   * @throws IOException
+   */
+  @Override
+  public void write() throws IOException {
+    if (this.dataFiles == null || this.dataFiles.length == 0) {
+      throw new RuntimeException("'withAvroPath()' must be called to support loading avro files");
+    }
+    for (CarbonFile dataFile : this.dataFiles) {
+      this.loadSingleFile(dataFile);
+    }
+  }
+
+  private void loadSingleFile(CarbonFile file) throws IOException {
```

Comment:
validateAndSetDataFiles itself can take single file (list of one element), why again another interface

Reply:
handle same for all the places

Reply:
Here we are loading the datafiles iteratively. In Invalidate method we are just confirming the files of same type and having the same schema.